### PR TITLE
Fix failure when reading non-alphanumeric column names

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Fixes https://github.com/trinodb/trino/issues/21881

## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# General
* TBD. ({issue}`issuenumber`)
```

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/sql/planner/SymbolAllocator.java [hunk 1]
```diff
@@ -61,6 +61,9 @@ public Symbol newSymbol(String nameHint, Type type)
         requireNonNull(type, "type is null");
 
         nameHint = EXCLUDED_CHARACTERS.trimAndCollapseFrom(nameHint.toLowerCase(ENGLISH), '_');
+        if (nameHint.isEmpty()) {
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/sql/planner/SymbolAllocator.java:64]: Add a check in Symbol constructor that provided name is not empty.

(if we allowed symbols named "", we should allow Variables named "")

Comment (findepi) [reply core/trino-main/src/main/java/io/trino/sql/planner/SymbolAllocator.java:64]: nit: Maybe move this just before line
```
Symbol symbol = new Symbol(type, nameHint);
```

File: core/trino-main/src/test/java/io/trino/sql/planner/TestSymbolAllocator.java [hunk 2]
```diff
@@ -36,4 +36,18 @@ public void testUnique()
 
         assertThat(symbols.size()).isEqualTo(4);
     }
+
+    @Test
+    public void testUnicode()
```
Comment (findepi) [inline core/trino-main/src/test/java/io/trino/sql/planner/TestSymbolAllocator.java:41]: "testNonAscii"?

**Review by findepi — APPROVED** [review]

**Comment by ebyhr** [comment]
CI hit #21724

**Pull request merged**